### PR TITLE
feat: add data seeding

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [ ] I have performed preliminary testing using the [test suite](../../apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
- [ ] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/OsmoX.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

Add data seeding file to run at the time of migration

**Related changes:**

- Add data seeding for the following tables
  1. notify_users
  2. notify_applications
  3. notify_server_api_keys
  4. notify_master_providers
- Add down method for them as well
- Add a new enum to define `ProviderTypes`
- Config data is as follows:

**Regex sources**

- Host URL: https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/RegEx.md
- Phone: https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/RegEx.md 
- Domain: https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html
- bcrypt Password: https://stackoverflow.com/questions/31417387/regular-expression-to-find-bcrypt-hash
- API Key: https://stackoverflow.com/questions/8571501/how-to-check-whether-a-string-is-base64-encoded-or-not/8571649#8571649
- Auth Token: custom generated
- Regex verification: https://regexr.com

**Screenshots:**

501migration
migration

501mp
master providers

**Query request and response:**

NA

**Documentation changes:**

NA

**Test suite output:**

NA

**Pending actions:**

NA

**Additional notes:**

NA

## Summary by CodeRabbit

- **New Features**
	- Added new provider types constants for EMAIL, SMS, and WHATSAPP.
	- Introduced auto-generated primary keys for database fields to enhance data integrity.
	- Implemented sample data seeding for notification providers, applications, and user credentials to facilitate testing and development.

- **Bug Fixes**
	- Improved error handling in the authentication service to provide clearer error messages during login failures.

- **Refactor**
	- Removed the `applicationId` field from the application creation process to simplify data input requirements.

### CODE REVIEW

File: apps/api/src/database/migrations/1692870736646-seeddata.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    await queryRunner.query(
+      `INSERT INTO notify_applications (application_id,name,user_id,created_on,updated_on,status) VALUES (1001,'sampleOsmoXApp',1,'2024-04-12 11:36:02','2024-04-12 11:36:02',1)`,
+    );
+
+    await queryRunner.query(
+      `INSERT INTO notify_server_api_keys (api_key,application_id,created_on,updated_on,status) VALUES ('OsmoX-test-key',1001,'2024-04-12 12:28:06','2024-04-12 12:28:06',1)`,
+    );
+
+    await queryRunner.query(
+      `INSERT INTO notify_master_providers (name,provider_type,configuration,created_on,updated_on,status) VALUES ('SMTP',1,'{"ENABLE_SMTP":{"label":"ENABLE SMTP","id":"ENABLE_SMTP","pattern":"/^(true|false)$/","type":"boolean"},"SMTP_HOST":{"label":"SMTP HOST","id":"SMTP_HOST","pattern":"^https?://(?:[^/]+|(?:\\d{1,3}\\.){3}\\d{1,3})/","type":"string"},"SMTP_PORT":{"label":"SMTP PORT","id":"SMTP_PORT","pattern":"\\d{1,5}","type":"number"},"SMTP_USERNAME":{"label":"SMTP USERNAME","id":"SMTP_USERNAME","pattern":"^[a-zA-Z0-9_-]{3,16}$","type":"string"},"SMTP_PASSWORD":{"label":"SMTP PASSWORD","id":"SMTP_PASSWORD","pattern":"^[a-fA-F0-9]{64}$","type":"string"}}','2024-04-24 15:20:23','2024-04-24 15:20:23',1),
```
Comment (LakshayaT) [inline apps/api/src/database/migrations/1692870736646-seeddata.ts:21]: For SMTP_HOST URL, you can use the pattern from here: https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/RegEx.md if it is fine.
Also I feel like SMTP_USERNAME upperlimit is too low.

File: apps/api/src/database/migrations/1692870736646-seeddata.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+      ('MAILGUN',1,'{"ENABLE_MAILGUN":{"label":"ENABLE MAILGUN","id":"ENABLE_MAILGUN","pattern":"/^(true|false)$/","type":"boolean"},"MAILGUN_API_KEY":{"label":"MAILGUN API KEY","id":"MAILGUN_API_KEY","pattern":"^[a-zA-Z0-9-_]{16,512}$","type":"string"},"MAILGUN_HOST":{"label":"MAILGUN HOST","id":"MAILGUN_HOST","pattern":"^https?://(?:[^/]+|(?:\\d{1,3}\\.){3}\\d{1,3})/","type":"string"},"MAILGUN_DOMAIN":{"label":"MAILGUN DOMAIN","id":"MAILGUN_DOMAIN","pattern":"^(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\\.)+[a-zA-Z]{2,}$","type":"string"}}','2024-04-24 15:26:10','2024-04-24 15:26:10',1),
```
Comment (LakshayaT) [inline apps/api/src/database/migrations/1692870736646-seeddata.ts:23]: Same here, can use our standards URL for host.

File: apps/api/src/database/migrations/1692870736646-seeddata.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      ('WA_360_DAILOG',3,'{"ENABLE_WA360DIALOG":{"label":"ENABLE WA360DIALOG","id":"ENABLE_WA360DIALOG","pattern":"/^(true|false)$/","type":"boolean"},"WA_360_DIALOG_URL":{"label":"MAILGUN API KEY","id":"MAILGUN_API_KEY","pattern":"^{https://waba.360dialog.io/v1/messages}$","type":"string"},"WA_360_DIALOG_API_KEY":{"label":"WA 360 DIALOG API KEY","id":"WA_360_DIALOG_API_KEY","pattern":"^[a-zA-Z0-9-_]{16,512}$","type":"string"}}','2024-04-24 15:31:54','2024-04-24 15:31:54',1),
+
+      ('WA_TWILIO',3,'{"ENABLE_WA_TWILIO":{"label":"ENABLE WA TWILIO","id":"ENABLE_WA_TWILIO","pattern":"/^(true|false)$/","type":"boolean"},"TWILIO_WA_ACCOUNT_SID":{"label":"TWILIO WA ACCOUNT SID","id":"TWILIO_WA_ACCOUNT_SID","pattern":"^AC\\w{3,}$","type":"string"},"TWILIO_WA_AUTH_TOKEN":{"label":"TWILIO WA AUTH TOKEN","id":"TWILIO_WA_AUTH_TOKEN","pattern":"^[a-zA-Z0-9-_]{16,512}$","type":"string"},"TWILIO_WA_NUMBER":{"label":"TWILIO WA NUMBER","id":"TWILIO_WA_NUMBER","pattern":"^\\d{10}$","type":"string"}}','2024-04-24 15:36:46','2024-04-24 15:36:46',1),
```
Comment (LakshayaT) [inline apps/api/src/database/migrations/1692870736646-seeddata.ts:27]: The mobile numbers can include country codes as well. Add that to the pattern too.

File: apps/api/src/database/migrations/1692870736646-seeddata.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+      ('SMS_TWILIO',2,'{"ENABLE_SMS_TWILIO":{"label":"ENABLE SMS TWILIO","id":"ENABLE_SMS_TWILIO","pattern":"/^(true|false)$/","type":"boolean"},"TWILIO_SMS_ACCOUNT_SID":{"label":"TWILIO SMS ACCOUNT SID","id":"TWILIO_SMS_ACCOUNT_SID","pattern":"^AC\\w{3,}$","type":"string"},"TWILIO_SMS_AUTH_TOKEN":{"label":"TWILIO SMS AUTH TOKEN","id":"TWILIO_SMS_AUTH_TOKEN","pattern":"^[a-zA-Z0-9-_]{16,512}$","type":"string"},"TWILIO_SMS_NUMBER":{"label":"TWILIO SMS NUMBER","id":"TWILIO_SMS_NUMBER","pattern":"^\\d{10}$","type":"string"}}','2024-04-24 15:42:15','2024-04-24 15:42:15',1);`,
```
Comment (LakshayaT) [inline apps/api/src/database/migrations/1692870736646-seeddata.ts:29]: Same here, update mobile number pattern to have country code too.

**Review by LakshayaT — COMMENTED** [review]
Please check the added comments

File: apps/api/src/database/migrations/1692870736646-seeddata.ts (hunk 1 above)
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: Fixed this, upper limit moved to 30

File: apps/api/src/database/migrations/1692870736646-seeddata.ts (hunk 3 above)
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: Fixed this

File: apps/api/src/database/migrations/1692870736646-seeddata.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    );
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: These queries are part of a seed data migration and intended to be run once only. Is this still an issue?

[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: Fixed in latest pr

[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: @coderabbitai check please

File: apps/api/src/modules/auth/auth.service.ts [hunk 6]
```diff
@@ -44,7 +44,7 @@ export class AuthService {
         allKeys: tokenList,
       };
     } catch (error) {
-      throw new error(`Error while logging in:\n${error}`);
+      throw new Error(`Error while logging in: ${error}`);
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/api/src/modules/auth/auth.service.ts]: Now only returning error.message

File: apps/api/src/database/migrations/1692870736646-seeddata.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (LakshayaT) [reply apps/api/src/database/migrations/1692870736646-seeddata.ts]: @coderabbitai This is resolved in https://github.com/OsmosysSoftware/osmo-x/pull/217/commits/01a1c05b4edf08fbdca84ccb9b6d8a9af67badc9

**Review by LakshayaT — APPROVED** [review]

**Pull request merged**